dataprep/clean/clean_date.py

```diff
@@ -0,0 +1,996 @@
+        parsed_target_format_data = check_target_format(target_format=target_format)
+        parsed_target_format_data = set_timezone_offset(target_timezone, parsed_target_format_data)
+        if parsed_target_format_data.valid:
+            if parsed_date_data.valid == "cleaned":
+                transformed_date = transform(
+                    parsed_date_data=parsed_date_data,
+                    parsed_target_format_data=parsed_target_format_data,
+                    target_format=target_format,
+                    target_timezone=target_timezone,
+                )
+                row[f"{col}_clean"] = f"{transformed_date}"
+                if row[col] != row[f"{col}_clean"]:
+                    STATS["cleaned"] += 1
+            else:
+                STATS["unknown"] += 1
+                row[f"{col}_clean"] = np.nan
+        else:
+            raise ValueError(
+                f"target_format {target_format} is invalid. "
+                f"Invalid tokens are {parsed_target_format_data.invalid_tokens}. "
+                f"Please retype it."
+            )
+    return row
+
+
+def set_timezone_offset(
+    timezone: Union[str, Any], parsed_data: Union[ParsedDate, ParsedTargetFormat, Any]
+) -> Any:
+    """
+    This function set timezone information for parsed date or parsed target format
+    Parameters
+    ----------
+    timezone
+        string name of timezone
+    parsed_data
+        parsed date or parsed target format
+    """
+    example_date = datetime.datetime(2009, 9, 1)
+    if timezone in all_timezones:
+        days, seconds = 0, 0
+        pytz_offset = pytz.timezone(timezone).utcoffset(example_date)
+        if not pytz_offset is None:
+            days = pytz_offset.days
+            seconds = pytz_offset.seconds
+        parsed_data.set_tzinfo(utc_offset_hours=int(abs(days) * 24 + abs(seconds) / 3600))
+        parsed_data.set_tzinfo(
+            utc_offset_minutes=int((abs(seconds) - (abs(seconds) / 3600) * 3600) / 60)
+        )
+        if days >= 0 and seconds >= 0:
+            parsed_data.set_tzinfo(utc_add="+")
+        elif days <= 0 and seconds < 0:
+            parsed_data.set_tzinfo(utc_add="-")
+    elif timezone in ZONE:
+        parsed_data.set_tzinfo(utc_offset_hours=abs(ZONE[timezone]))
+        parsed_data.set_tzinfo(utc_offset_minutes=0)
+        if ZONE[timezone] >= 0:
+            parsed_data.set_tzinfo(utc_add="+")
+        elif ZONE[timezone] < 0:
+            parsed_data.set_tzinfo(utc_add="-")
+    return parsed_data
+
+
+def validate_date(date: Union[str, pd.Series]) -> Union[bool, pd.Series]:
+    """
+    This function validates url
+    Parameters
+    ----------
+    date
+        pandas Series of urls or url instance
+    """
+    if isinstance(date, pd.Series):
+        verfied_series = date.apply(check_date)
+        return verfied_series
+    else:
+        return check_date(date)
+
+
+def validate_target_format(target_format: Union[str, Any]) -> Any:
```

Comment:
Is it redundant? If it is, we can remove it

Reply:
You mean the two space lines? If yes, after black's formatting, it generates the two space lines. Without this kind of format, I can't pass the static check.

Reply:
I mean the `validate_target_format()`. I think it's not ever called in the code?

Reply:
Yeah. I've refined it.